src/ioctls/vcpu.rs

```diff
@@ -972,6 +972,30 @@ impl VcpuFd {
         Ok(())
     }
 
+    /// Returns the value of the specified vCPU register.
+    ///
+    /// The id of the register is encoded as specified in the kernel documentation
+    /// for `KVM_SET_ONE_REG`.
+    ///
+    /// # Arguments
+    ///
+    /// * `reg_id` - ID of the register.
+    ///
+    #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
+    pub fn get_one_reg(&self, reg_id: u64) -> Result<u64> {
```

Comment:
This ioctl behaves (sort of) the same as `kvm_get_irqchip` and `kvm_get_msrs`. To keep the kvm-ioctls interface consistent (and as described in the KVM API documentation) this function should receive a mutable `kvm_one_reg` structure and return the register in that structure and the error code in the result.

For example, this is the signature of `get_irqchip`:
```rust
pub fn get_irqchip(&self, irqchip: &mut kvm_irqchip) -> Result<()>
```

Reply:
The structure **kvm_one_reg** doesn't actually contain any output data, and it is not written to by the kvm. The result of the KVM_GET_ONE_REG call is an u64 value  and will be written to the address pointed to by **kvm_one_reg**.**addr**. 

Based on the above I believe the current interface is better since it does not put any overhead of filling the kvm_one_reg struct on the user.

Reply:
I kind of like the model where the ioctl wrapper does the heavy work (working with the kernel structs) as well.

In retrospect this is what should've been done for `get_irqchip` as well. I will open a RFC to that end 😄

Reply:
Let's leave it as is right now. It's also a very subjective thing.